wp-pdf-templates.php

```diff
@@ -207,24 +207,47 @@ function _use_pdf_template() {
 
       if( defined('FETCH_COOKIES_ENABLED') && FETCH_COOKIES_ENABLED ) {
 
-        // we want a html template
-        $header = 'Accept:text/html' . "\n";
-
-        // pass cookies from current request
-        if( isset( $_SERVER['HTTP_COOKIE'] ) ) {
-          $header .= 'Cookie: ' . $_SERVER['HTTP_COOKIE'] . "\n";
-        }
-
-        // create a request context for file_get_contents
-        $context = stream_context_create(array(
-          'http' => array(
-            'method' => 'GET',
-            'header' => $header,
-          )
-        ));
-
-        // load the generated html from the template endpoint
-        $html = file_get_contents($link . (strpos($link, '?') === false ? '?' : '&') . 'pdf-template', false, $context);
+		$url =  $link . (strpos($link, '?') === false ? '?' : '&') . 'pdf-template';
+		$ch = curl_init();
+		
+		curl_setopt($ch, CURLOPT_URL, $url);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+		
+		// get http headers for cookies
+		curl_setopt($ch, CURLOPT_VERBOSE, 1);
+		curl_setopt($ch, CURLOPT_HEADER, 1);
+		
+		// store the cookies
+		$cookies = array();
+		foreach ($_COOKIE as $key => $value)
+		{
+			if ($key != 'Array')
+			{
+				$cookies[] = $key . '=' . $value;
+			}
+		}
+		// add the cookies to the request
+		curl_setopt( $ch, CURLOPT_COOKIE, implode(';', $cookies) );
+		
+		// Stop session so curl can use the same session without conflicts
+		session_write_close();
+		
+		$response = curl_exec($ch);
+		curl_close($ch);
+		
+		// Session restart
+		session_start();
```

Comment:
I guess that may be the case, since we're only performing one operation with this session. If we were doing more than one thing we'd want to unlock the session with session_write_close(). See: http://blog.psych0tik.net/2012/12/a-look-at-php-session-write-locks/comment-page-1/